output/stationRetrospective.sh

```diff
@@ -0,0 +1,104 @@
+# files with fullpath in reverse chronological order on
+# standard input.
+#-----------------------------------------------------------------------
+# path="/mnt/nas-storage/Operations/fortytwo.cct.lsu.edu/2022/nam/202204????/HSOFS/qbc.loni.org/HSOFS_nam_akheir/nowcast"
+# fileList=$(printf "\"%s\",\n" $(ls -d $path | sort))
+# printf "{ \"fort.61.nc\" : [ ${fileList::-1} ] }" | json_pp > fileList.json
+#-----------------------------------------------------------------------
+# april2022=( $(find /mnt/nas-storage/Operations/fortytwo.cct.lsu.edu/2022/nam -name LAv20a -print | grep 202204) )
+# for p in ${april2022[@]}; do echo $p ; nowcastDirs+=( $(find $p -name nowcast -print) ) ; done
+# rm -f tmp ; for d in ${nowcastDirs[@]} ; do echo $d/fort.61.nc >> tmp ; done ; cat tmp | sort -r > filelist.txt
+# stationRetrospective.sh < filelist.txt
+# -----------------------------------------------------------------------
+# cat ../../../../Operations/autodownloader_config_april_nowcast.inventory | grep hsofs-nam-bob-2021 | grep fort.61.nc | sort -r > shortpath_filelist.txt
+# awk '{ print "/mnt/nas-storage/Operations/"$0 }' shortpath_filelist.txt > filelist.txt
+# stationRetrospective.sh < filelist.txt
+# -----------------------------------------------------------------------
+# cat ../../../Operations/autodownloader_config_april_nowcast.inventory | grep NCSC | grep fort.61.nc | sort -r > shortpath_filelist.txt
+# awk '{ print "/mnt/nas-storage/Operations/"$0 }' shortpath_filelist.txt > filelist.txt
+# stationRetrospective.sh < filelist.txt
+# -----------------------------------------------------------------------
+# cat /mnt/nas-storage/Operations/autodownloader_config_april_nowcast.inventory | grep hsofs-nam-bob-2021 | grep fort.72.nc | sort -r > fort.72.shortpath_filelist.txt
+# awk '{ print "/mnt/nas-storage/Operations/"$0 }' fort.72.shortpath_filelist.txt > fort.72.filelist.txt
+# stationRetrospective.sh < fort.72.filelist.txt
+# -----------------------------------------------------------------------
+# export RETROSPECTIVE_TARGET_DATE=2022050100  # set end date for the retrospective
+# -----------------------------------------------------------------------
+THIS=$(basename -- $0)
+timePeriods=( 1 2 4 7 14 21 30 ) # days back from the last available date
+timePeriodCount=0
+dataSetCount=0
+targetEpochSeconds=0
+declare -a catList  # must ncrcat in chronological order
+declare -a timePeriodsSeconds
+for p in ${timePeriods[@]}; do
+    timePeriodsSeconds+=( $(( $p * 86400 )) )
+done
+while read fname
+do
+    #echo "$fname"
+    fn=$(basename $fname)
+    fileList+=( $fname )
+    catList=( $fname ${catList[@]} )
+    base_date=$(ncdump -h $fname | grep base_date | cut -d = -f 2 | tr -d '";')
+    csEpochSeconds=$(TZ=UTC date -u -d "$base_date" "+%s")
+    dataSetSecondsList=( $(ncks --json -v 'time' $fname | grep data | tr -d '"data:[,]') )
+    numDataSets=$(ncdump -h $fname | grep currently | grep -Eo [0-9]+)
```

Comment:
Do you think parsing the output of `ncdump` at this point should be a Perl script? Maybe if any more complexity is needed> Just a thought/question.

Reply:
I think we are in good shape for now, but we are getting close to the point where it is needed ... `jq` would be another good option